docs/data/tree-view/rich-tree-view/editing/editing.md

```diff
@@ -0,0 +1,44 @@
+---
+productId: x-tree-view
+title: Rich Tree View - Focus
+githubLabel: 'component: tree view'
+packageName: '@mui/x-tree-view'
+waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
+packageName: '@mui/x-tree-view'
+---
+
+# Rich Tree View - label editing
+
+<p class="description">Learn how to edit the label of Tree View items.</p>
+
+:::warning
+this doc page is stip WIP
+:::
+
+## Enable editing on certain items
+
+You can use the `isItemEditable` prop to enable editing on certain items.
+
+If an item is editable, the editing state can be toggled by double clicking on it, or by pressing <kbd class="key">Enter</kbd> on the keyboard when the item is in focus.
+
+Once an item is in editing state, the value of the label can be edited. Pressing <kbd class="key">Enter</kbd> again or bluring the item will save the new value. Pressing <kbd class="key">Esc</kbd> will cancel the action and restore the item to its original state.
+
+{{"demo": "LabelEditing.js"}}
+
+## Change the default behavior
+
+The default behavior on bluring the tree item that is being edited is to save the new value if there is one. This can be customized using the `slotProps` of the `TreeItem2`
+
+{{"demo": "CustomBehavior.js"}}
+
+## Limit editing to leaves
```

Comment:
I would make this section a child of `## Enable editing on certain items`

Reply:
> We could consider dropping the interactions in favor of only using publicAPI methods at some point

I think that makes sense. 

> I'm not entirely sure how the DX should look like at this level...

We could have a [similar DX](https://mui.com/x/react-data-grid/api-object/#inside-the-data-grid) to the grid, but it's slightly confusing IMO :(

Reply:
We already return `publicAPI` from `useTreeItem2` :+1: